modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -1340,3 +1324,143 @@ export async function signRecoveryMpcV2(
+): Promise<{
+  userKeyShare: Buffer;
+  backupKeyShare: Buffer;
+  commonKeyChain: string;
+}> {
+  const userCompressedPrv = Buffer.from(sjcl.decrypt(walletPassphrase, encryptedMPCv2UserKey), 'base64');
+  const bakcupCompressedPrv = Buffer.from(sjcl.decrypt(walletPassphrase, encryptedMPCv2BackupKey), 'base64');
+
+  const userPrvJSON: DklsTypes.ReducedKeyShare = DklsTypes.getDecodedReducedKeyShare(userCompressedPrv);
+  const backupPrvJSON: DklsTypes.ReducedKeyShare = DklsTypes.getDecodedReducedKeyShare(bakcupCompressedPrv);
+  const userKeyRetrofit: DklsTypes.RetrofitData = {
+    xShare: {
+      x: Buffer.from(userPrvJSON.prv).toString('hex'),
+      y: Buffer.from(userPrvJSON.pub).toString('hex'),
+      chaincode: Buffer.from(userPrvJSON.rootChainCode).toString('hex'),
+    },
+    xiList: userPrvJSON.xList.slice(0, 2),
+  };
+  const backupKeyRetrofit: DklsTypes.RetrofitData = {
+    xShare: {
+      x: Buffer.from(backupPrvJSON.prv).toString('hex'),
+      y: Buffer.from(backupPrvJSON.pub).toString('hex'),
+      chaincode: Buffer.from(backupPrvJSON.rootChainCode).toString('hex'),
+    },
+    xiList: backupPrvJSON.xList.slice(0, 2),
+  };
+  const [user, backup] = await DklsUtils.generate2of2KeyShares(userKeyRetrofit, backupKeyRetrofit);
+  const userKeyShare = user.getKeyShare();
+  const backupKeyShare = backup.getKeyShare();
+  const commonKeyChain = DklsTypes.getCommonKeychain(userKeyShare);
+  return { userKeyShare, backupKeyShare, commonKeyChain };
+}
+
+/**
+ * Gets the combined key for GG18
+ * @param encryptedGG18UserKey encrypted GG18 user key
+ * @param encryptedGG18BackupKey encrypted GG18 backup key
+ * @param walletPassphrase wallet passphrase
+ * @returns key shares
+ */
+function getKeyCombinedFromTssKeyShares(
+  encryptedGG18UserKey: string,
+  encryptedGG18BackupKey: string,
+  walletPassphrase?: string
+): [ECDSAMethodTypes.KeyCombined, ECDSAMethodTypes.KeyCombined] {
+  let backupPrv;
+  let userPrv;
+  try {
+    backupPrv = sjcl.decrypt(walletPassphrase, encryptedGG18BackupKey);
+    userPrv = sjcl.decrypt(walletPassphrase, encryptedGG18UserKey);
+  } catch (e) {
+    throw new Error(`Error decrypting backup keychain: ${e.message}`);
+  }
+
+  const userSigningMaterial = JSON.parse(userPrv) as ECDSAMethodTypes.SigningMaterial;
+  const backupSigningMaterial = JSON.parse(backupPrv) as ECDSAMethodTypes.SigningMaterial;
+
+  if (!userSigningMaterial.backupNShare) {
+    throw new Error('Invalid user key - missing backupNShare');
+  }
+
+  if (!backupSigningMaterial.userNShare) {
+    throw new Error('Invalid backup key - missing userNShare');
+  }
+
+  const MPC = new Ecdsa();
+
+  const userKeyCombined = MPC.keyCombine(userSigningMaterial.pShare, [
+    userSigningMaterial.bitgoNShare,
+    userSigningMaterial.backupNShare,
+  ]);
+  const backupKeyCombined = MPC.keyCombine(backupSigningMaterial.pShare, [
+    backupSigningMaterial.userNShare,
+    backupSigningMaterial.bitgoNShare,
+  ]);
```

Comment:
Note: no derivation path is applied here.

Reply:
signRecoveryMpcv2 signs for m/0 all the time, which makes this work. However, I think that function's name should probably change to reflect that behaviour (signRecoveryMpcv2ForBaseAddress) or a parameter could be added to allow different derivation paths to be passed with a default of m/0.